implementation/sequicity/tsd_net.py

```diff
@@ -92,11 +92,60 @@ def nan(v):
 
 def get_sparse_input_aug(x_input_np):
     """
-    sparse input of
-    :param x_input_np: [T,B]
-    :return: Numpy array: [B,T,aug_V]
+    :param x_input_np: Numpy (int?) array: [T,B]
+    :return: Pytorch float tensor: [B,T,aug_V]
+
+    Where:
+        - T = maximum length of sequence
+        - B = batch size
+        - aug_V = (vocab_size + T) = augmented vocabulary = vocabulary + input sentence words
+
+    Additional needed inputs:
+      - `vocab_size` is currently read from cfg.vocab_size.
+      - TODO: I personally don't like global variables, I believe it would be
+        nicer to have `vocab_size` at the input of this function.
+
+    Input x_input_np:
+        - Integer-encoded sentences
+        - t=0 (the first time index) is ignored.
+            - TODO: Where is this mentioned in CopyNet?
+            - This seems to be "right-shifting", as known from Transformer
+              architecture: If it weren't here, the model would just learn to
+              always copy the input to the output.
+        - Example: [[0,0,0], [3,2,1],[1,0,0]] contains three sentences:
+            - 1st sentence is of length 2: (word3 word1)
+            - 2nd sentence is of length 1: (word2)
+            - 3rd sentence is of length 1: (word1)
+        - TODO: I'd personally prefer examples, where all dimensions/variables
+          have different values. eg. vocab_size=5, T=4, B=3. This is 3x3, so T
+          = B, not a nice example
+
+    Result result:
+        - One-hot-encoded sentences
+        - Converted input example:
+            - [
+                [[0,0,0,0,1,0,0], [0,0,1,0,0,0,0]],
+                [[0,0,0,1,0,0,0], [0,0,0,0,0,0,0]],
+                [[0,0,1,0,0,0,0], [0,0,0,0,0,0,0]],
+            ]
+            - where those zeros are actually 10^-9 (TODO: Why is this?)
+            - TODO: What is vocab_size in this example?
+            - TOOD: I assume word0 is not possible to have. Why is (word1) not [1,0,0,0,0,0,0], but [0,0,1,0,0,0,0]?
```

Comment:
word0 is possible, sorry, should have made it clear.
The additional 0 at the beggining is because of ignoring index 0.
However, I'll review the example in more depth to answer your questions, I might have made a mistake.